### PR TITLE
Integrate swagger and add env api

### PR DESCRIPTION
**Pull request by majian159**
issues #763
1. install swag
```console
$ go get -u github.com/swaggo/swag/cmd/swag
```
2. run `make swagger-gen`
```console
swag init -g route.go -d pkg/server/ -o docs/swagger/
2020/12/10 17:29:26 Generate swagger docs....
2020/12/10 17:29:26 Generate general API Info, search dir:pkg/server/
2020/12/10 17:29:26 Generating apis.Environment
2020/12/10 17:29:26 Generating apis.Response
2020/12/10 17:29:26 Generating apis.EnvironmentBody
2020/12/10 17:29:26 create docs.go at docs/swagger/docs.go
2020/12/10 17:29:26 create swagger.json at docs/swagger/swagger.json
2020/12/10 17:29:26 create swagger.yaml at docs/swagger/swagger.yaml

# output
docs/swagger/
├── docs.go
├── swagger.json
└── swagger.yaml
```

### CODE REVIEW

File: docs/swagger/swagger.json [hunk 1]
```diff
@@ -0,0 +1,429 @@
+{
```
Comment (zzxwill) [inline docs/swagger/swagger.json:1]: 严格意义上来讲，这个不算 doc，你移到 `pkg/server/docs` 下面哈，我也是依赖这个路径生成 openapi html.

File: pkg/server/route.go [hunk 2]
```diff
@@ -12,7 +12,11 @@ import (
 )
 
 // setup the gin http server handler
+// @title KubeVela API
+// @version 0.0.1
+// @description An KubeVela API.
 
+// @BasePath /api
```
Comment (zzxwill) [inline pkg/server/route.go:19]: 你的 merge 之后，我会再增加一些内容。

File: charts/vela-core/crds/core.oam.dev_applicationdeployments.yaml [hunk 3]
```diff
@@ -21,42 +21,53 @@ spec:
   - name: v1alpha2
     schema:
       openAPIV3Schema:
-        description: ApplicationDeployment is the Schema for the ApplicationDeployment API
+        description: ApplicationDeployment is the Schema for the ApplicationDeployment
+          API
```
Comment (zzxwill) [inline charts/vela-core/crds/core.oam.dev_applicationdeployments.yaml:25]: Why is the description changed?

File: pkg/server/route.go [hunk 4]
```diff
@@ -136,6 +145,11 @@ func (s *APIServer) setupRoute(staticPath string) http.Handler {
 
 	// version
 	api.GET(util.VersionPath, s.GetVersion)
+
+	// swagger
+	url := ginSwagger.URL("/swagger/doc.json")
+	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler, url))
```
Comment (zzxwill) [inline pkg/server/route.go:151]: Cool!

**Comment by hongchaodeng** [comment]
Please resolve the conflict files

**Comment by majian159** [comment]
> Please resolve the conflict files

resolved

File: Makefile [hunk 5]
```diff
@@ -226,3 +226,7 @@ endif
 start-dashboard:
 	go run pkg/server/main/startAPIServer.go &
 	cd dashboard && yarn && yarn start && cd ..
+	cd dashboard && npm install --registry=https://registry.npm.taobao.org && npm start && cd ..
```
Comment (hongchaodeng) [inline Makefile:229]: Isn't this duplicate to above command?

**Review by hongchaodeng — APPROVED** [review]
lgtm after nit

File: pkg/server/route.go [hunk 6]
```diff
@@ -136,6 +145,11 @@ func (s *APIServer) setupRoute(staticPath string) http.Handler {
 
 	// version
 	api.GET(util.VersionPath, s.GetVersion)
+
+	// swagger
+	url := ginSwagger.URL("/swagger/doc.json")
```
Comment (hongchaodeng) [inline pkg/server/route.go:150]: Why is this `doc.json`?

**Comment by hongchaodeng** [comment]
@majian159 Please fix above comments. Thanks!

**Comment by hongchaodeng** [comment]
Thanks @majian159

**Pull request merged**